s64da_benchmark_toolkit/prepare.py

```diff
@@ -176,18 +176,32 @@ def _load_pre_schema(self, conn):
                 conn.cursor.execute(pre_schema_file.read())
 
     def _load_license(self, conn):
-        license_path = '/s64da.license'
         try:
-            conn.cursor.execute(f'select swarm64da.load_license(\'{license_path}\')')
-            print(f'Loading S64 DA License {license_path}')
+            conn.cursor.execute(f'select swarm64da.show_license()')
+        except errors.InvalidSchemaName:
+            print(f'Selected schema is not for S64 DA. Skip checking license.')
+            return
+        except errors.UndefinedFunction:
+            print(f'License check function not required or supported. Skip checking license.')
+            return
         except:
-            print(f'Could not find a license at {license_path}. Skip loading license.')
+            pass
 
-        print(f'S64 DA License Status:')
         try:
-            self._run_shell_task(f'psql {self.args.dsn} -c "select swarm64da.show_license()"')
+            license_path = '/s64da.license'
+            try:
+                conn.cursor.execute(f'select swarm64da.load_license(\'{license_path}\')')
+                print(f'Loading S64 DA License {license_path}')
+            except:
+                print(f'Could not find a license at {license_path}. Skip loading license.')
+
+            print(f'S64 DA License Status:')
+            try:
+                self._run_shell_task(f'psql {self.args.dsn} -c "select swarm64da.show_license()"')
+            except:
+                print(f'No license file loaded or license invalid.')
         except:
-            print(f'No license file loaded or license invalid.')
+            return
```

Comment:
I am fine with the approach, but please make it more linear, no nested exceptions. Maybe only a single try-except block with error resolutions as needed.